lighthouse/helpers/plates.py

```diff
@@ -470,21 +495,174 @@ def create_cherrypicked_post_body(barcode: str, samples: List[Dict[str, Any]]) -
+            content["supplier_name"] = sample["supplier_name"]
             content["control"] = sample["control"]
             content["control_type"] = sample["control_type"]
+            content["uuid"] = sample["uuid"]
         else:
             content["name"] = sample["name"]
             content["phenotype"] = sample["phenotype"]
             content["supplier_name"] = sample["supplier_name"]
             content["sample_description"] = sample["sample_description"]
+            content["uuid"] = sample["uuid"]
 
         wells_content[sample["coordinate"]] = {"content": content}
 
+    subjects = []
+    subjects.append(robot_subject(robot_serial_number))
+    subjects.extend(source_plate_subjects(plate_id_mappings))
+    subjects.extend(sample_subjects(samples))
+
+    events = [
+        {
+            "event": {
+                "user_identifier": user_id,
+                "event_type": PLATE_EVENT_DESTINATION_CREATED,
+                "subjects": subjects,
+                "metadata": {},
+                "lims": app.config["RMQ_LIMS_ID"],
+            }
+        }
+    ]
+
     body = {
         "barcode": barcode,
         "purpose_uuid": app.config["SS_UUID_PLATE_PURPOSE_CHERRYPICKED"],
         "study_uuid": app.config["SS_UUID_STUDY_CHERRYPICKED"],
         "wells": wells_content,
+        "events": events,
     }
 
     return {"data": {"type": "plates", "attributes": body}}
+
+
+def sample_subjects(samples):
+    subjects = []
+    for sample in samples:
+        if "control" in sample:
+            subject = {
+                "role_type": "control",
+                "subject_type": "sample",
+                "friendly_name": control_friendly_name(sample),
+                "uuid": sample["uuid"],
+            }
+        else:
+            subject = {
+                "role_type": "sample",
+                "subject_type": "sample",
+                "friendly_name": sample_friendly_name(sample),
+                "uuid": sample["uuid"],
+            }
+        subjects.append(subject)
+    return subjects
+
+
+def sample_friendly_name(sample):
+    name = "__".join(
+        [sample["sample_description"], sample["name"], sample["lab_id"], sample["phenotype"]]
+    )
+    return name
+
+
+def control_friendly_name(sample):
+    return f"{sample['supplier_name']}"
+
+
+# def destination_labware_subject(barcode):
+#     subject = {
+#         "role_type": "cherrypicking_destination_labware",
+#         "subject_type": "plate",
+#         "friendly_name": barcode,
+#     }
+#     return subject
+
+
+def source_plate_subjects(plate_id_mappings):
+    subjects = []
+    for mapping in plate_id_mappings:
+        subject = {
+            "role_type": "cherrypicking_source_labware",
+            "subject_type": "plate",
+            "friendly_name": mapping["barcode"],
+            "uuid": mapping["uuid"],
+        }
+        subjects.append(subject)
+    return subjects
+
+
+def robot_subject(robot_serial_number):
```

Comment:
Again could use `helpers/plate_events` methods for this

Reply:
I'm thinking a good future (no need to do here) change would be to have a single method that all the event subject generation methods call, so that we only define the "role_type", "subject_type" etc. keys in a single place

Reply:
To be done in future refactoring on a different branch